### DISCUSSION TITLE
A force as a BC for a node

### DISCUSSION THREAD

**Discussion by AlexSacMorane** [idaholab/moose#21068] [Q&A Modules: Solid mechanics] [answered]
Hello everyone,
I am looking to apply some force on a point as boundaries condition. I have seen it is possible to use external pressure on my solid, but I prefer a punctual condition. Is it possible? 
You will find my mesh and my .i in the following :
[Moose_test.zip](https://github.com/idaholab/moose/files/8755571/Moose_test.zip)
Thanks in advance for any help
I stay available
Regards
Alexandre

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think you can do it.
Create a nodeset with only 1 point in it and specify it as the `boundary` parameter of the dirichletBCs to apply the load.

Guillaume

**Reply by AlexSacMorane** [reply]
I have added a node set (id : 13) and some boundary condition on stress_xx for this node but I get an error message : 

```
*** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".

No nonlinear variable named stress_xx found. Did you specify an auxiliary variable when you meant to specify a nonlinear variable (or vice-versa)?
```

I generate this variable with the following : 

```
[Modules/TensorMechanics/Master]
  [./all]
    strain = FINITE
    planar_formulation = PLANE_STRAIN
    add_variables = true
    generate_output = 'stress_xx stress_xy stress_yy stress_zz strain_xx strain_xy strain_yy strain_zz'
  [../]
[]
```

I drop the new version of the mesh/.i
[Test_Moose.zip](https://github.com/idaholab/moose/files/8761084/Test_Moose.zip)
Thanks

**Reply by GiudGiud** [reply]
I m not sure you can do that, a BC on stress, like this.
Can you turn on show_var_residual_norm in The [Debug] block when running the case without this boundary condition to see the list of variables

**Reply by AlexSacMorane** [reply]
Thanks, I will take a look at the debug block.
And do you know if there is another way to apply an external mechanical force ?
Regards
Alexandre

**Reply by lynnmunday** [reply]
You can't apply stress_xx as the variable for a dirichlet bc.  The variable for a bc means the variable you are solving for, so it would be disp_x if you want to apply a force in the x direction.  Then you should apply one of the neumann bc's if you want a force.  There might be some examples of using tractions in the tensor mechanics module if you want to use stress.

You also are not using automatic differentiation variables so you need change your bc from ADDirichletBC to just DirichletBC.

**Reply by AlexSacMorane** [reply]
Thanks for this answer I will look for this
